.github/workflows/verify-build.yml

```diff
@@ -37,6 +37,8 @@ jobs:
     name: Unit Tests
     needs: basic-checks
     runs-on: ${{ github.repository == 'hyperledger/fabric' && 'fabric-ubuntu-20.04' || 'ubuntu-20.04' }}
+    env:
+      GOEXPERIMENT: nocoverageredesign
```

Comment:
What exactly breaks if we don't add GOEXPERIMENT=nocoverageredesign ?

Reply:
https://github.com/golang/go/issues/67427

```
--- FAIL: TestEndorsementPlugin (3.03s)
panic: Error opening plugin at path /var/folders/6v/kmxqr0yn61588pbz646cfdyc0000gq/T/TestEndorsementPlugin4080854438/001/endorsementplugin.so: plugin.Open("/var/folders/6v/kmxqr0yn61588pbz646cfdyc0000gq/T/TestEndorsementPlugin4080854438/001/endorsementplugin"): plugin was built with a different version of package github.com/hyperledger/fabric/core/handlers/endorsement/api [recovered]
        panic: Error opening plugin at path /var/folders/6v/kmxqr0yn61588pbz646cfdyc0000gq/T/TestEndorsementPlugin4080854438/001/endorsementplugin.so: plugin.Open("/var/folders/6v/kmxqr0yn61588pbz646cfdyc0000gq/T/TestEndorsementPlugin4080854438/001/endorsementplugin"): plugin was built with a different version of package github.com/hyperledger/fabric/core/handlers/endorsement/api
```

Reply:
`go test -v -cover github.com/hyperledger/fabric/core/handlers/endorsement/api github.com/hyperledger/fabric/core/handlers/library`

Reply:
from github action:

```
2024-05-15 16:03:16.284 UTC 0001 PANI [core.handlers] loadPlugin -> Error opening plugin at path /tmp/TestEndorsementPlugin3801377298/001/endorsementplugin.so: plugin.Open("/tmp/TestEndorsementPlugin3801377298/001/endorsementplugin"): plugin was built with a different version of package github.com/hyperledger/fabric/core/handlers/endorsement/api
--- FAIL: TestEndorsementPlugin (2.23s)
panic: Error opening plugin at path /tmp/TestEndorsementPlugin3801377298/001/endorsementplugin.so: plugin.Open("/tmp/TestEndorsementPlugin3801377298/001/endorsementplugin"): plugin was built with a different version of package github.com/hyperledger/fabric/core/handlers/endorsement/api [recovered]
	panic: Error opening plugin at path /tmp/TestEndorsementPlugin3801377298/001/endorsementplugin.so: plugin.Open("/tmp/TestEndorsementPlugin3801377298/001/endorsementplugin"): plugin was built with a different version of package github.com/hyperledger/fabric/core/handlers/endorsement/api

goroutine 34 [running]:
testing.tRunner.func1.2({0x19efbc0, 0xc00041a5a0})
	/opt/hostedtoolcache/go/1.22.3/x64/src/testing/testing.go:1631 +0x3f7
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.22.3/x64/src/testing/testing.go:1634 +0x6b6
panic({0x19efbc0?, 0xc00041a5a0?})
	/opt/hostedtoolcache/go/1.22.3/x64/src/runtime/panic.go:770 +0x132
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x2, 0xc000166820, {0x4a?, 0xc000264300?, 0x17b?})
	/home/runner/work/fabric/fabric/vendor/go.uber.org/zap/zapcore/entry.go:196 +0x9f
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc000166820, {0x0, 0x0, 0x0})
	/home/runner/work/fabric/fabric/vendor/go.uber.org/zap/zapcore/entry.go:262 +0x3ad
go.uber.org/zap.(*SugaredLogger).log(0xc0002b8e70, 0x4, {0xc000276360, 0x117}, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0})
	/home/runner/work/fabric/fabric/vendor/go.uber.org/zap/sugar.go:316 +0x130
go.uber.org/zap.(*SugaredLogger).Panicf(...)
	/home/runner/work/fabric/fabric/vendor/go.uber.org/zap/sugar.go:202
github.com/hyperledger/fabric-lib-go/common/flogging.(*FabricLogger).Panicf(...)
	/home/runner/work/fabric/fabric/vendor/github.com/hyperledger/fabric-lib-go/common/flogging/zap.go:74
github.com/hyperledger/fabric/core/handlers/library.(*registry).loadPlugin(0xc000443e18, {0xc00003a840, 0x3d}, 0x2, {0xc000443e98, 0x1, 0x1})
	/home/runner/work/fabric/fabric/core/handlers/library/plugin.go:30 +0x3a9
github.com/hyperledger/fabric/core/handlers/library.TestEndorsementPlugin(0xc00037d040)
	/home/runner/work/fabric/fabric/core/handlers/library/registry_plugin_test.go:100 +0x1e7
testing.tRunner(0xc00037d040, 0x1c[215](https://github.com/pfi79/fabric/actions/runs/9098906093/job/25010318274#step:5:216)88)
	/opt/hostedtoolcache/go/1.22.3/x64/src/testing/testing.go:1689 +0x21f
created by testing.(*T).Run in goroutine 1
	/opt/hostedtoolcache/go/1.22.3/x64/src/testing/testing.go:1742 +0x826
FAIL	github.com/hyperledger/fabric/core/handlers/library	31.645s
```

Reply:
Have you tried to resolve this test failure without the workaround? Or is that what you are asking in the go community? I wouldn't want to use a workaround which may go away in the next release. But maybe you are suggesting this is a temporary Go issue that will get resolved by next release?

Reply:
> Have you tried to resolve this test failure without the workaround? Or is that what you are asking in the go community? I wouldn't want to use a workaround which may go away in the next release. But maybe you are suggesting this is a temporary Go issue that will get resolved by next release?

Yes I have tried various ways to solve the problem, assuming the error is somewhere in our fabric. But so far my conclusion is that the error is in go.

I went through all the questions in the go community on this topic and realized the following:
- they redesigned the cover tool.
- There are a lot of comments on the new cover. The developers themselves recognize this and are waiting for fixes.
- so they introduced the GOEXPERIMENT=nocoverageredesign flag.

What happens next:
- some fixes to the new cover will be made. I don't know to what extent. 
- At the end GOEXPERIMENT=nocoverageredesign will be canceled.

I hope that by that time in new go releases our problem will be solved.
I highlighted with my question that their new cover created a problem for us. I will wait and hope that the developers will see and respond to my question. Maybe they will offer some other alternative solution.